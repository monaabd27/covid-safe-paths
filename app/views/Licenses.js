import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import {
  BackHandler,
  Image,
  Linking,
  ScrollView,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';

import fontFamily from './../constants/fonts';
import { Images } from '../assets';
import NavigationBarWrapper from '../components/NavigationBarWrapper';
import { Typography } from '../components/Typography';
import Colors from '../constants/colors';
import { Theme } from '../constants/themes';
import { config } from '../COVIDSafePathsConfig';

const PRIVACY_POLICY_URL =
  'https://docs.google.com/document/d/17u0f8ni9S0D4w8RCUlMMqxAlXKJAd2oiYGP8NUwkINo/edit';

export const LicensesScreen = ({ navigation }) => {
  const { t } = useTranslation();

  const backToMain = () => {
    navigation.goBack();
  };

  const handleBackPress = () => {
    backToMain();
    return true;
  };

  const handleTermsOfUsePressed = () => {
    Linking.openURL(PRIVACY_POLICY_URL);
  };

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', handleBackPress);
    return () => {
      BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
    };
  });

  const legalPageHeaderText =
    config.tracingStrategy === 'gps'
      ? t('label.legal_page_header_location')
      : t('label.legal_page_header_bluetooth');

  return (
    <NavigationBarWrapper
      title={t('label.legal_page_title')}
      onBackPress={backToMain}>
      <ScrollView contentContainerStyle={styles.contentContainer}>
        <View>
          <Typography style={styles.heading} use='headline2'>
            {legalPageHeaderText}
          </Typography>
          <Typography style={styles.body} use='body1'>
            {t('label.legal_page_address')}
          </Typography>
          <Typography
            style={styles.hyperlink}
            onPress={() => {
              Linking.openURL('mailto:info@pathcheck.org');
            }}>
            {/* eslint-disable-next-line react-native/no-raw-text */}
            {'info@pathcheck.org'}
          </Typography>
          <Typography
            style={styles.hyperlink}
            onPress={() => {
              Linking.openURL('https://covidsafepaths.org/');
            }}>
            {/* eslint-disable-next-line react-native/no-raw-text */}
            {'covidsafepaths.org'}
          </Typography>
        </View>
      </ScrollView>

      <Theme use='charcoal'>
        <TouchableOpacity
          onPress={handleTermsOfUsePressed}
          style={styles.termsInfoRow}>
          <Typography
            use='body1'
            onPress={() => Linking.openURL(PRIVACY_POLICY_URL)}>
            {t('label.privacy_policy')}
          </Typography>
          <View style={styles.arrowContainer}>
            <Image source={Images.ForeArrow} />
          </View>
        </TouchableOpacity>
      </Theme>
    </NavigationBarWrapper>
  );
};

const styles = StyleSheet.create({
  body: {
    color: Colors.BLACK,
    marginBottom: 24,
  },
  contentContainer: {
    width: '100%',
    backgroundColor: Colors.INTRO_WHITE_BG,
    paddingHorizontal: 26,
    paddingVertical: 40,
  },
  heading: {
    color: Colors.BLACK,
    marginBottom: 12,
  },
  hyperlink: {
    color: Colors.VIOLET_TEXT,
    fontSize: 16,
    marginBottom: 12,
    fontFamily: fontFamily.primaryRegular,
    textDecorationLine: 'underline',
  },
  termsInfoRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: Colors.VIOLET,
    paddingVertical: 25,
    paddingHorizontal: 15,
  },
  arrowContainer: {
    alignSelf: 'center',
    paddingRight: 20,
    paddingLeft: 20,
  },
});
