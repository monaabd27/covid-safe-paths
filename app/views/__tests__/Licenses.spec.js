import 'react-native';

import { render } from '@testing-library/react-native';
import React from 'react';

import { LicensesScreen } from '../Licenses';

jest.mock('../../COVIDSafePathsConfig', () => {
  return {
    config: { tracingStrategy: 'gps' },
  };
});

it('renders correctly', () => {
  const { asJSON } = render(<LicensesScreen />);

  expect(asJSON()).toMatchSnapshot();
});
