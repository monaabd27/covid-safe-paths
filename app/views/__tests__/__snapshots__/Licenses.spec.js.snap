// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#4051DB",
        "flexBasis": 0,
        "flexGrow": 0,
        "flexShrink": 1,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F7F8FF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4051DB",
          "borderBottomColor": "#5061E6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "alignItems": "center",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": 60,
            "zIndex": 1,
          }
        }
      >
        <SvgXml
          style={
            Object {
              "color": "#FFF",
              "height": 18,
              "width": 18,
            }
          }
          xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 16,
            "letterSpacing": 1,
            "lineHeight": 34,
            "paddingHorizontal": 20,
            "position": "absolute",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "100%",
          }
        }
      >
        Legal
      </Text>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "#F7F8FF",
          "paddingHorizontal": 26,
          "paddingVertical": 40,
          "width": "100%",
        }
      }
    >
      <View>
        <View>
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 26,
                "fontWeight": "bold",
                "lineHeight": 34,
                "marginBottom": 12,
                "writingDirection": "ltr",
              }
            }
            use="headline2"
          >
            PathCheck GPS
          </Text>
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "IBMPlexSans",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "marginBottom": 24,
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            PathCheck
58 Day Street
Box 441621
Somerville, MA 02144
          </Text>
          <Text
            style={
              Object {
                "color": "#3C4ED8",
                "fontFamily": "IBMPlexSans",
                "fontSize": 16,
                "fontWeight": "normal",
                "lineHeight": 24,
                "marginBottom": 12,
                "textDecorationLine": "underline",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            info@pathcheck.org
          </Text>
          <Text
            style={
              Object {
                "color": "#3C4ED8",
                "fontFamily": "IBMPlexSans",
                "fontSize": 16,
                "fontWeight": "normal",
                "lineHeight": 24,
                "marginBottom": 12,
                "textDecorationLine": "underline",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            covidsafepaths.org
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "backgroundColor": "#4051DB",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "IBMPlexSans",
            "fontSize": 18,
            "fontWeight": "normal",
            "lineHeight": 24,
            "writingDirection": "ltr",
          }
        }
        use="body1"
      >
        Privacy Policy
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/images/foreArrow.png",
            }
          }
        />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
